tools/apiview/parsers/apiview-treestyle-parser-schema/models.tsp

```diff
@@ -0,0 +1,150 @@
+  BottomTokens?: StructuredToken[];
+
+  /** The nodes immediate children. For a namespace this would be classes, for a class this would be the class constructors and methods. Children are rendered after TopTokens but before BottomTokens, and are automatically indented.*/
+  Children?: APITreeNode[];
+
+  /** Properties of the APITreeNode. */
+  Properties?: APITreeNodeProperties;
+
+  /** Tags of the APITreeNode. */
+  Tags?: APITreeNodeTags[];
+}
+
+model StructuredToken {
+  /** Token Id. Previously known as DefinitionId. */
+  Id: string;
+
+  /** Represents the type of a structured token*/
+  Kind: StructuredTokenKind;
+
+  /** The token value which will be displayed. Spacing tokens (LineBreak, NonBreakingSpace, TabSpace, and ParameterSeparator) don't need to have value */
+  Value?: string;
+
+  /** Properties of the StructuredToken. */
+  Properties?: StructuredTokenProperties;
+
+  /** Tags of the StructuredToken. */
+  Tags?: StructuredTokenTags[];
+
+  /** Classes used to render the StructuredTokens. */
+  RenderClasses?: StructuredTokenRenderClasses[]
+}
+
+model APITreeNodeProperties {
+  /** Use this only if you are looking to add a custom icon different from language wide defaults set by APITreeNode kind*/
+  IconName?: string;
+}
+
+model StructuredTokenProperties {
+  /** Group a sequence of consecutive StructuredTokens. doc indicates that the group of tokens is documentation. */
+  GroupId?: "doc";
+
+  /** Id to navigate to when StructuredToken is clicked. */
+  NavigateToId?:  string;
+}
+
+enum StructuredTokenKind {
+  /** Default kind */
+  Content,
+
+  /** Starts a new line */
+  LineBreak,
+
+  /** Single Space */
+  NonBreakingSpace,
+
+  /** Four spaces */
+  TabSpace,
+  
+  /** Use this between method parameters. Depending on user setting this would result in a LineBreak or a LineBreak + TabSpace */
+  ParameterSeparator
+}
+
+/** Classes used to render the StructuredTokens. Extendable by contributing styles at https://github.com/Azure/azure-sdk-tools/blob/main/src/dotnet/APIView/ClientSPA/src/app/_components/code-panel/code-panel.component.scss */
+union StructuredTokenRenderClasses {
+  "comment",
+  "keyword",
+  "keyword",
+  "literal",
+  /** string literal */
+  "sliteral",
+  /** member name */
+  "mname",
+  /** type name */
+  "tname",
+  /** punctuation */
+  "punc",
+  "text",
+  string
+}
+
+union APITreeNodeTags {
+  /** Mark a token as deprecated. */
+  "Deprecated",
+  /** Mark a node as Hidden. */
+  "Hidden",
+  /** Indicate that a node should be hidden from the page navigation. */
+  "HideFromNav",
+  /** Indicate that a node should not be used in computation of diff. */
+  "SkipDiff",
+  /** The cross language definitionId for the node. */
+  "CrossLangDefId"
+}
+
+union StructuredTokenTags {
+  /** Mark a token as deprecated */
+  "Deprecated",
+  /** Indicate that a Token should not be used in computing diff*/
+  "SkipDiff",
+}
+
+enum CodeDiagnosticLevel {
+  Default,
```

Comment:
It's unusual in a server to have "default" as an actual level. What does "default" represent compared to "info"?

Reply:
@praveenkuttappan do you know the answer to this?

Reply:
We don't need the DEFAULT enum. Older token style had this default because we initially had only one type of diagnostic for system comments(not categorized as error or warning) and then there were 3 more different types added later.

We can modify this in new parser to have `info, warning, error and fatal`. 

What is the type of system generated comments in new .NET parser. If that's tagged as `DEFAULT` then we can update the parser to categorize them correctly with the help of .NET team or if they are okay to treat all of the current ones as ERROR then we can update the parser accordingly.